### ISSUE TITLE
Possibility to use DocBlock for arrays without FQCN

### ISSUE THREAD

**Issue by morrislaptop** [spatie/data-transfer-object#161]
If we were to use `Symfony\Component\PropertyInfo\Extractor\PhpDocExtractor` we wouldn't have to use a FQCN in docblock comments.

Would you be interested in a PR?

**Comment by mfn** [comment]
See https://github.com/spatie/data-transfer-object/pull/14 , there was an attempt and discussion but the add complexity also due to necessity for caching was a factor for not proceeding (at that time, anyway).

**Comment by morrislaptop** [comment]
Ah I see, maybe the symfony package does this in a smarter way with caching included?

**Comment by brendt** [comment]
I don't think it's worth adding this feature, since it goes against docblock standards, adds complexity and because most types can already be represented with built-in types. We're also going to drop docblock type check altogether in the next major.

**Comment by morrislaptop** [comment]
No problem. What’s the plan for typing arrays in the next major without
docblock support? Always use collection classes?

**Comment by brendt** [comment]
The ideas are outlined here: https://github.com/spatie/data-transfer-object/issues/151

tl;dr: focus on static analysis instead of runtime type checks.
